### PR TITLE
Improve `get_demo` dataset 

### LINKED ISSUE #374: Create a more descriptive demo dataset
### Problem Description

Right now the demo dataset for RDTs includes a variety of column types, but (a) the context is very abstract and (b) there is no boolean column.

```python
from rdt import get_demo
get_demo()
```
image

For demos and documentation, it would be better to have a dataset that is more contextual for the users to understand.

### Expected behavior

Return the following dataset instead:

image

```python
from rdt import get_demo

# returns a dataset with 5 rows
data = get_demo()

# returns a dataset 100 rows
data = get_demo(num_rows=100)
```

### Additional context

The first 5 rows of the dataset should be hardcoded. The code below generates them.

```python
import numpy as np
import pandas as pd

login_dates = ['2021-06-26', '2021-02-10', 'NAT', '2020-09-26', '2020-12-22']
last_login = [np.datetime64(i) for i in login_dates]

email_optin = [False, False, False, True, np.nan]
credit_card = ['VISA', 'VISA', 'AMEX', np.nan, 'DISCOVER']
age = [29, 18, 21, 45, 32]
dollars_spent = [99.99, np.nan, 2.50, 25.00, 19.99]

df = pd.DataFrame(data={
    'last_login': last_login,
    'email_optin': email_optin,
    'credit_card': credit_card,
    'age': age,
    'dollars_spent': dollars_spent
})
```

An additional rows can be randomly generated according to the following rules:
- Set a seed so we always return the same rows
- `age` can never be `NaN`
- `credit_card` can be either `"VISA"`, `"AMEX"`, `"DISCOVER"` or `NaN`
- `dollars_spent` cannot have more than 2 decimal digits

### PR DESCRIPTION
**Pull request by fealho**
Resolve #374.

### CODE REVIEW

**Comment by fealho** [comment]
@npatki Example data generated by the new `get_demo`:
```
   last_login  email_optin credit_card  age  dollars_spent
0  2021-06-26        False        VISA   29          99.99
1  2021-02-10        False        VISA   18            NaN
2         NaT        False        AMEX   21           2.50
3  2020-09-26         True         NaN   45          25.00
4  2020-12-22         <NA>    DISCOVER   32          19.99
5  2019-11-27         True         nan   61          22.02
6  2002-05-10         <NA>        AMEX   41          71.11
7  2014-10-04         <NA>    DISCOVER   96          80.95
8  2014-03-19        False        AMEX   76          34.87
9         NaT         True        AMEX   49           9.62
10 2017-02-25         <NA>        VISA   69          94.05
11 2001-04-11        False        AMEX   79          39.76
12 2012-02-13         <NA>        VISA   75          51.78
13 2015-04-10         <NA>        VISA   69          83.77
14 2022-10-14         True        AMEX   29          67.57
15 2004-08-12         <NA>    DISCOVER   56          73.52
16 2002-02-08         True    DISCOVER   19            NaN
17 2019-01-10         <NA>    DISCOVER   20          54.14
18        NaT        False    DISCOVER   73            NaN
19 2014-07-17         <NA>    DISCOVER   98          22.86
20 2013-10-30         True        AMEX   76          17.50
21        NaT         True        AMEX   19          98.22
22 2003-03-30        False         nan   19          51.66
23 2012-06-21         <NA>    DISCOVER   71            NaN
24 2009-04-08         <NA>        AMEX   18          99.63
25 2017-07-06        False         nan   36          96.54
26        NaT         <NA>    DISCOVER   19          55.83
27        NaT         <NA>        VISA   70          88.26
28 2007-01-02         True         nan   61          18.87
29 2021-06-28         <NA>        AMEX   49            NaN
30 2005-08-09         <NA>        VISA   87          70.04
31        NaT        False        VISA   49          84.67
32 2025-02-05        False        VISA   85            NaN
33 2027-05-17         True         nan   72          40.45
34 2000-07-08         <NA>        AMEX   92            NaN
35 2007-06-27         <NA>        VISA   73          85.09
36 2008-03-24         <NA>    DISCOVER   34          93.56
37 2012-10-02         True        VISA   55          78.53
38 2005-03-14         True        VISA   41            NaN
39 2021-03-05        False         nan   86          58.07
40 2003-06-21         True         nan   87          37.23
41 2004-03-08         <NA>        AMEX   28          94.01
42 2009-09-26         <NA>    DISCOVER   33          97.37
43 2010-08-26         True         nan   90          28.39
44 2024-03-13         <NA>        VISA   76          30.54
45        NaT         <NA>        AMEX   87          48.56
46 2024-01-27         True        VISA   97            NaN
47        NaT         True        VISA   20            NaN
48        NaT         <NA>         nan   37          17.59
49 2007-02-25         <NA>        AMEX   76           1.81
```

File: tests/unit/test___init__.py [hunk 1]
```diff
@@ -4,6 +4,17 @@
...
-    assert len(demo) == 10
-    assert list(demo.isna().sum(axis=0)) == [2, 2, 2, 2]
+    assert list(demo.columns) == [
+        'last_login', 'email_optin', 'credit_card', 'age', 'dollars_spent'
+    ]
+    assert len(demo) == 5
+    assert list(demo.isna().sum(axis=0)) == [1, 1, 1, 0, 1]
+
+
+def test_get_demo_many_rows():
```
Comment (npatki) [inline tests/unit/test___init__.py:14]: Could we also add a test to make that the same rows are returned every time? -- ie validate that fixing the seed is working for everything we generate

**Review by npatki — APPROVED** [review]
Requested a test to ensure the seed is working -- and that we don't accidentally get rid of it in the future. Other than that, LGTM!

File: rdt/__init__.py [hunk 2]
```diff
@@ -33,26 +33,46 @@ def get_demo(dtypes=('int', 'float', 'str', 'datetime'), nans=0.2, size=10):
...
+        'credit_card': credit_card,
+        'age': age,
+        'dollars_spent': dollars_spent
+    })
+
+    if num_rows <= 5:
+        return data.iloc[:num_rows]
+
+    # Randomly generate the remaining rows
+    np.random.seed(42)
```
Comment (pvk-developer) [inline rdt/__init__.py:56]: Why do we use seed ?

**Review by pvk-developer — APPROVED** [review]

File: rdt/__init__.py (hunk 2 above)
Comment (fealho) [reply rdt/__init__.py:56]: From the issue: "_Set a seed so we always return the same rows_"

**Pull request merged**